src/morphology.rs

```diff
@@ -511,13 +514,11 @@ impl Mask {
     /// # }
     /// ```
     pub fn diamond(radius: u8) -> Self {
-        Self {
-            elements: (-(radius as i16)..=(radius as i16))
-                .flat_map(|x| {
-                    ((x.abs() - radius as i16)..=(radius as i16 - x.abs())).map(move |y| (x, y))
-                })
-                .collect(),
-        }
+        let mut elements = Vec::with_capacity(1 + 2 * (radius as usize) * (radius as usize + 1));
+        elements.extend((-(radius as i16)..=(radius as i16)).flat_map(|y| {
```

Comment:
Add `let radius = i16::from(radius);` above

Reply:
done, as well as in other functions where the problem appeared